assets/js/src/core/modules/data-object/tree/hooks/use-node-api-hook.ts

```diff
@@ -46,10 +46,13 @@ export const useNodeApiHook = (node: TreeNodeProps): NodeApiHookReturnType => {
 
     const dataObjectData = data.items
     dataObjectData.forEach((dataObjectNode) => {
-      const icon = dataObjectNode.icon?.type === 'name' ? dataObjectNode.icon?.value : 'mainObject'
+      const icon = dataObjectNode.icon?.value === '' || dataObjectNode.icon?.value === 'vector'
```

Comment:
We have some more complexity in there. In the response from the server (for both assets and data-objects) we can get an icon via 'customAttributes.icon'. If an icon is defined there it will overrule the regular icon. The format of the icon is the same like for the regular one (with type + icon properties). The background for this case are our events. When the icon is overruled via event the change will be added to the customAttributes.